pkg/asset/manifests/topologies.go

```diff
@@ -33,7 +34,7 @@ func determineTopologies(installConfig *types.InstallConfig) (controlPlaneTopolo
 		// Two node deployments with 0 workers mean that the control plane nodes are treated as workers
 		// in that situation we have decided that it is appropriate to set the infrastructureTopology to HA.
 		// All other configuration for different worker count are respected with the original intention.
-		if controlPlaneReplicas == 2 {
+		if controlPlaneTopology == configv1.DualReplicaTopologyMode || controlPlaneTopology == configv1.HighlyAvailableArbiterMode {
```

Comment:
When `controlPlaneTopology == configv1.SingleReplicaTopologyMode`, `infrastructureTopology`  should also be `configv1.SingleReplicaTopologyMode`. Is there a specific check needed here for when worker count != 0, or is this case being handled by an install config validation?

Reply:
We kept the behavior here the same as before, we just carved out an exception for Arbiter and DualReplica when worker count will be 0 but we need the infra to still be HA. In SingleReplica you can have workers so in the event of single replica it follows the path that was already present, 0 worker: `infrastructureTopology=SingleReplicaTopologyMode`, 1 worker: `infrastructureTopology=SingleReplicaTopologyMode` and n worker: `infrastructureTopology=HAMode`